src/spikeinterface/sorters/internal/si_based.py

```diff
@@ -14,8 +14,16 @@ def is_installed(cls):
 
     @classmethod
     def _setup_recording(cls, recording, output_folder, params, verbose):
-        # nothing to do here because the spikeinterface_recording.json is here anyway
-        pass
+        # Some recording not json serializable but they can be saved to pickle
+        #   * NoiseGeneratorRecording or InjectTemplatesRecording: we force a pickle because this is light
+        #   * for NumpyRecording (this is a bit crazy because it flush the entire buffer!!)
+        if recording.check_if_dumpable() and not isinstance(recording, NumpyRecording):
+            rec_file = output_folder.parent / "spikeinterface_recording.pickle"
+            recording.dump_to_pickle(rec_file)
```

Comment:
@samuelgarcia remove these changes, they will be handled in #2027

Reply:
yep. but i need this PR to merged first

Reply:
ok done